python/ray/dag/input_node.py

```diff
@@ -57,19 +58,36 @@ class InputNode(DAGNode):
     ...     # dag_input["m1"], dag_input["m2"]
     """
 
-    def __init__(self, *args, _other_args_to_resolve=None, **kwargs):
+    def __init__(
+        self,
+        *args,
+        input_types: Dict[Union[int, str], type] = None,
+        _other_args_to_resolve=None,
+        **kwargs,
+    ):
         """InputNode should only take attributes of validating and converting
         input data rather than the input data itself. User input should be
         provided via `ray_dag.execute(user_input)`.
 
         Args:
+            input_types: Map of key -> type. Describes the type of each
+                InputAttributeNode. Used to determine through what type of Gradio block
```

Comment:
> Used to determine through what type of Gradio block the InputAttributeNode should be displayed in the deployment graph visualization.

s/w Used when deciding what Gradio block to represent the InputAttributeNode with.